runtime/src/main/java/org/corfudb/runtime/view/stream/IStreamView.java

```diff
@@ -0,0 +1,122 @@
+package org.corfudb.runtime.view.stream;
+
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Function;
+
+/** This interface represents a view on a stream. A stream is an ordered
+ * set of log entries which can only be appended to and read in sequential
+ * order.
+ *
+ * Created by mwei on 1/5/17.
+ */
+public interface IStreamView extends Iterator<LogData> {
+
+    /** Return the ID of the stream this view is for.
+     * @return  The ID of the stream.
+     */
+    UUID getStreamID();
+
+    /** Reset the state of this stream view, causing the next read to
+     * start from the beginning of this stream.
+     */
+    void reset();
+
+    /** Append an object to the stream, returning the physical address
+     * it was written at.
+     * <p>
+     * Optionally, provide a method to be called when an address is acquired,
```

Comment:
change this: Optionally, provide a method to be injected when an <address, whatever we call it> is acquired for this append, before the entry is actually written. This is necessary for "Compare-and-Swap" type of operations.